From 1a37cd929bcc2c387c23cee1cdcfffd6d9863116 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Mon, 13 Jun 2011 20:32:18 -0400
Subject: [PATCH] daemon: don't set LANG to empty value

This makes PAM not set it to a proper value later.

https://bugzilla.gnome.org/show_bug.cgi?id=649413
---
 daemon/gdm-session-direct.c |   18 ++++++++++++------
 1 files changed, 12 insertions(+), 6 deletions(-)

Index: gdm-3.0.4/daemon/gdm-session-direct.c
===================================================================
--- gdm-3.0.4.orig/daemon/gdm-session-direct.c	2011-05-31 17:46:56.000000000 +0200
+++ gdm-3.0.4/daemon/gdm-session-direct.c	2011-10-14 14:30:55.039767843 +0200
@@ -1962,6 +1962,8 @@
 static void
 setup_session_environment (GdmSessionDirect *session)
 {
+        const char *locale;
+
         gdm_session_direct_set_environment_variable (session,
                                                      "GDMSESSION",
                                                      get_session_name (session));
@@ -1969,12 +1971,16 @@
                                                      "DESKTOP_SESSION",
                                                      get_session_name (session));
 
-        gdm_session_direct_set_environment_variable (session,
-                                                     "LANG",
-                                                     get_language_name (session));
-        gdm_session_direct_set_environment_variable (session,
-                                                     "GDM_LANG",
-                                                     get_language_name (session));
+        locale = get_language_name (session);
+
+        if (locale != NULL && locale[0] != '\0') {
+                gdm_session_direct_set_environment_variable (session,
+                                                             "LANG",
+                                                             locale);
+                gdm_session_direct_set_environment_variable (session,
+                                                             "GDM_LANG",
+                                                             locale);
+        }
 
         gdm_session_direct_set_environment_variable (session,
                                                      "DISPLAY",
